dagmc/dagnav.py

```diff
@@ -1,12 +1,46 @@
+"""
+This file includes the DAGModel module which is designed to manage and 
+manipulate hierarchical geometric data structures used in computational 
+modeling and simulation, particularly in the context of Discrete Geometry. 
+It uses the PyMOAB library to handle geometric entities, ensuring efficient
+access, manipulation, and storage of geometric data.
+"""
+
 from __future__ import annotations
 from abc import abstractmethod
 from functools import cached_property
 from itertools import chain
 from typing import Optional, Dict
 from warnings import warn
-
+import logging
 import numpy as np
-from pymoab import core, types, rng, tag
+
+try:
+    from pymoab import core, types, rng, tag
+except ImportError as e:
```

Comment:
I know I suggested this in #1, but the message is a overly verbose for my taste. There are (somewhat dated) instructions on how to build MOAB with PyMOAB enabled [here](https://ftp.mcs.anl.gov/pub/fathom/moab-docs/building.html). I think we should point to them in this message and update the installation instructions there.

Also, while we may want to add a logging capability to this project in the future, I don't see the point of importing that module for this single message. We can raise our own message from a `ModuleNotFoundError` 

```python

try:
    from pymoab import core, types, rng, tag
except ImportError as e:
    msg = 'PyMOAB package was not found, please install this package using the instructions found here: https://ftp.mcs.anl.gov/pub/fathom/moab-docs/building.html'
    raise ModuleNotFoundError(msg) from e
```